src/contracts/pods/EigenPod.sol

```diff
@@ -488,10 +550,20 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
     }
 
 
+    /**
+     * @notice Called by the pod owner to activate restaking by withdrawing 
+     * all existing ETH from the pod and preventing further withdrawals via 
+     * "withdrawBeforeRestaking()"
+    */ 
+    function activateRestaking() external onlyEigenPodOwner hasNeverRestaked {
+        mostRecentWithdrawalTimestamp = uint32(block.timestamp);
+        hasRestaked = true;
+        _sendETH(podOwner, address(this).balance);
+    }
 
     /// @notice Called by the pod owner to withdraw the balance of the pod when `hasRestaked` is set to false
     function withdrawBeforeRestaking() external onlyEigenPodOwner hasNeverRestaked {
```

Comment:
prob make this logic an internal function an use it `withdrawBeforeRestaking` and `activateRestaking`